bgpd/bgp_table.c

```diff
@@ -149,49 +135,39 @@ void bgp_delete_listnode(struct bgp_node *node)
 		} else
 			return;
 
-		rn = bgp_node_to_rnode(node);
+		rn = bgp_dest_to_rnode(dest);
 
 		if (bgp && rn && rn->lock == 1) {
 			/* Delete the route from the selection pending list */
-			if ((node->rt_node)
+			if ((dest->rt_node)
 			    && (bgp->gr_info[afi][safi].route_list)) {
 				list_delete_node(
 					bgp->gr_info[afi][safi].route_list,
-					node->rt_node);
-				node->rt_node = NULL;
+					dest->rt_node);
+				dest->rt_node = NULL;
 			}
 		}
 	}
 }
 
-static struct bgp_node *
-bgp_route_next_until_maxlen(struct bgp_node *node, const struct bgp_node *limit,
+static struct route_node *
+bgp_route_next_until_maxlen(struct route_node *node,
+			    const struct route_node *limit,
 			    const uint8_t maxlen)
 {
-	const struct prefix *p = bgp_node_get_prefix(node);
-
-	if (node->l_left) {
-		const struct prefix *left_p =
-			bgp_node_get_prefix(bgp_node_from_rnode(node->l_left));
-
-		if (p->prefixlen < maxlen && left_p->prefixlen <= maxlen)
-			return bgp_node_from_rnode(node->l_left);
-	}
-
-	if (node->l_right) {
-		const struct prefix *right_p =
-			bgp_node_get_prefix(bgp_node_from_rnode(node->l_right));
+	if (node->l_left && node->p.prefixlen < maxlen
```

Comment:
looks like kinda partly revert of https://github.com/FRRouting/frr/pull/6085?

Reply:
I don't think this is there anymore?